src/utils/agent_tools.py

```
@@ -0,0 +1,1622 @@
+    async with ctx.db_lock:
+        for obs_id in observation_ids:
+            try:
+                await crud.delete_document(
+                    ctx.db,
+                    workspace_name=ctx.workspace_name,
+                    document_id=obs_id,
+                    observer=ctx.observer,
+                    observed=ctx.observed,
+                )
+                deleted_count += 1
+            except Exception as e:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)